scripts/cbmc_json_parser.py

```diff
@@ -0,0 +1,398 @@
+                    is difficult to parse and in some cases, is not in JSON format.
+
+    Returns - 
+        temp_dict - The restructured dictionary which is easy to parse
+
+    """
+    
+    # The restructured dictionary which is easy to parse
+    temp_dict = {}
+    try:
+        with open(log_file) as infile:
+            data = json.load(infile)
+            temp_dict["response"] = data
+    except FileNotFoundError:
+        print("CBMC File not found")
+        return {}
+    except JSONDecodeError:
+        # print out specific information about the extra data (like the lines before it)
+        print("JSON File not being parsed", )
+        return {}
+
+    # TODO: Check if passing dictionaries around is optimal or if there are alternate ways to 
+    # restructure and parse cbmc blob.
+    return temp_dict
+
+# Get Information about the Solver from the Json file written
+def extract_solver_information(json_object):
+    
+    # Solver information is all the fields in the json object which are not related to the results
+    solver_information = []
+    try:
+        responses = json_object["response"]
+    except:
+        print("No such Value called \"response\"")
+
+    # Check for the objects which are not related to the result object
+    for response_object in responses:
+        if "result" not in response_object.keys():
+            solver_information.append(response_object)
+        else:
+            return solver_information
+
+    return solver_information
+
+# From the extracted information, construct a message and append to the final Output
+def construct_solver_information_message(solver_information):
+    """
+    Construct a message and append to the final Output
+
+    Sample Output - 
+        CBMC 5.36.0 (cbmc-5.36.0)
+        CBMC version 5.36.0 (cbmc-5.36.0) 64-bit x86_64 linux
+        Reading GOTO program from file
+        ...
+    """
+    
+    solver_information_message = ""
+    for message_object in solver_information:
+        # 'Program' and 'messageText' fields give us the information about the solver
+        try:
+            if "program" in message_object.keys():
+                solver_information_message += message_object['program']
+            # Check message texts for valuable information and append only relevant messages
+            elif "messageText" in message_object.keys():
+                if message_object['messageText'] != 'Building error trace':
+                    solver_information_message += message_object['messageText']
+                else:
+                    pass
+            else:
+                solver_information_message += ''
+        except KeyError as e:
+            print("Key Error, Missing Properties in reading Solver Information from JSON")
+        solver_information_message += '\n'
+    return solver_information_message
+
+# Get property tests and results from the Json file written
+def extract_properties(json_object):
```

Comment:
This function body looks a lot like the function `extract_solver_information(json_object)`

Reply:
Will take out the common functionality into it's own function